excel.py

```diff
@@ -0,0 +1,56 @@
+#! python3
+# Reads an excel file and transforms to JSON.
+
+import xlrd
+from collections import OrderedDict
+import simplejson as json
+import psycopg2
+import urllib.parse
+
+# set up the connection. Extract the database info from the URI and connect with psycopg2
+def connect():
+    try:
+        result = urllib.parse.urlparse("postgres://4cebab44:4j602JWnaP3Yaw@db.nuodata.io:5432/4cebab44")
+        username = result.username
+        password = result.password
+        database = result.path[1:]
+        hostname = result.hostname
+        DB = psycopg2.connect(
+            database = database,
+            user = username,
+            password = password,
+            host = hostname
+        )
+        c = DB.cursor()
+        return DB, c
+    except:
+        print("<couldn't connect>")
+
+# opens the excel file, extracts data row by row, returns it in JSON
+def tojson():
+    # Open the workbook.
+    wb = xlrd.open_workbook('practice.xlsx')
```

Comment:
Much less reusable if you put plain constant strings within a function. make 'practice.xlsx' a function parameter.